pkg/kncloudevents/retries.go

```diff
@@ -180,3 +182,91 @@ func SelectiveRetry(_ context.Context, response *http.Response, err error) (bool
 	// Do Not Retry 1XX, 2XX, 3XX & Most 4XX StatusCode Responses
 	return false, nil
 }
+
+// generateBackoffFunction returns a valid retryablehttp.Backoff implementation which
+// wraps the provided RetryConfig.Backoff implementation with optional "Retry-After"
+// header support.
+func generateBackoffFn(config *RetryConfig) retryablehttp.Backoff {
```

Comment:
@creydr is there a reason why we are adding this function here?

Reply:
This was moved from `pkg/kncloudevents/message_sender_new.go` file and is still needed in the event_dispatcher: https://github.com/knative/eventing/blob/a39a3d582eb49c187f1bd96dc50754402f5dda8a/pkg/kncloudevents/event_dispatcher.go#L362